fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java

```diff
@@ -23,7 +24,8 @@ public class FoldConstantsRule extends BottomUpScalarOperatorRewriteRule {
 
     @Override
     public ScalarOperator visitCall(CallOperator call, ScalarOperatorRewriteContext context) {
-        if (call.isAggregate() || notAllConstant(call.getChildren())) {
+        if (call.isAggregate() || notAllConstant(call.getChildren()) ||
+                FunctionSet.ASSERT_TRUE.equals(call.getFnName())) {
```

Comment:
Should handle in ScalarOperatorEvaluator evaluation method.

Reply:
Ok

Reply:
done